codegen/module_system.go

```diff
@@ -134,6 +134,24 @@ func NewDefaultModuleSystem(
 		)
 	}
 
+	if err := system.RegisterClass("clients", module.Class{
```

Comment:
Yeah, I find it neat to be able to leverage the `SingleModule` type for the initialization piece. 👍 

Regarding the naming, I guess we can keep it as is for now since the config is one time only and doesn't really provide much information.